src/hotspot/share/gc/shared/ageTable.hpp

```diff
@@ -52,17 +53,18 @@ class AgeTable {
 
   // clear table
   void clear();
+  // check whether it's clear
+  bool is_clear() PRODUCT_RETURN0;
 
   // add entry
   inline void add(oop p, size_t oop_size);
 
   void add(uint age, size_t oop_size) {
-    assert(age > 0 && age < table_size, "invalid age of object");
```

Comment:
We now also include age 0 objects in the census for the case of concurrent collectors in the young generation. This slot would have `sizes[0] == 0` for our traditional use cases for generational stop-world collectors.

I'll upstream this change separately in a sibling PR/JBS. TBD.

Reply:
https://bugs.openjdk.org/browse/JDK-8314329 will track this and other changes to the age table that need to be separately udstreamed.